test/Interop/Cxx/namespace/templates-with-forward-decl-module-interface.swift

```diff
@@ -0,0 +1,9 @@
+// RUN: %target-swift-ide-test -print-module -module-to-print=TemplatesWithForwardDecl -I %S/Inputs -source-filename=x -enable-cxx-interop | %FileCheck %s
+
+// CHECK: extension NS1 {
+// CHECK:   struct __CxxTemplateInstN3NS14DeclIiEE {
+// CHECK:     typealias MyInt = Int32
+// CHECK:     var fwd: NS1.__CxxTemplateInstN3NS115ForwardDeclaredIiEE
```

Comment:
Can we check these class template specializations too? Might as well, these module interface tests are pretty easy.

Reply:
Amusingly, we would need https://github.com/apple/swift/pull/37695 for that :) Currently the lookup logic gets confused & only the contents of the first `NS1` redecl end up in the module interface. Applying https://github.com/apple/swift/pull/37695 locally solves this (all the contents get included into the module interface), but I guess it would be better to get this PR merged first, so it's too early to add a `// CHECK` for those decls now.

Reply:
Sounds good. Maybe if you think of it, you could fix this as a follow up. But it's not super important. The test is good as is.